multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ServiceBindingParametersGetter.java

```diff
@@ -92,8 +93,13 @@ private Map<String, Object> getFileProvidedBindingParameters(String requiredDepe
     }
 
     private Map<String, Object> getDescriptorProvidedBindingParameters(CloudApplicationExtended app, CloudServiceInstanceExtended service) {
-        return app.getBindingParameters()
-                  .getOrDefault(service.getResourceName(), Collections.emptyMap());
+        BindingDetails bindingDetails = app.getBindingParameters()
+                                           .get(service.getResourceName());
+        if (bindingDetails == null) {
+            return Collections.emptyMap();
+        }
+        context.setVariable(Variables.BINDING_NAME, bindingDetails.getBindingName());
+        return (bindingDetails.getBindingParameters() != null) ? bindingDetails.getBindingParameters() : Collections.emptyMap();
```

Comment:
this get method is changing the behavoiur of the flow which is not expected to do. 
The variable if needed must be set when the binding parameters are set in context

Reply:
The best way is to change:
Variable<Map<String, Object>> SERVICE_BINDING_PARAMETERS to <String, BindingDetails>]
this way you won't need to set separately the binding name from the binding details